core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -200,6 +202,7 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
     if (zkUtils.pathExists(toResourcePath(resource))) {
       zkUtils.deletePath(toResourcePath(resource))
       updateAclChangedFlag(resource)
+      updateCache(resource, Set())
```

Comment:
That would result in an infinite loop of change notifications. Since every change notification calls `updateCache`.